frontend/lib/identification/identification_qr_content_parser.dart

```diff
@@ -1,114 +1,88 @@
+import '../qr_code_scanner/qr_code_processor.dart';
 import 'base_card_details.dart';
 import 'card_details.dart';
 import 'card_details_model.dart';
 import 'protobuf/card_activate_model.pb.dart';
 
-QRCodeParseResult createFieldMissingError(
-    String missingFieldName, String publicErrorCode) {
-  return QRCodeParseResult(
-      hasError: true,
-      userErrorMessage: "Beim Verarbeiten des eingescannten Codes ist ein"
-          "Fehler aufgetreten. Fehlercode: $publicErrorCode");
+class QRCodeInvalidTotpSecretException extends QrCodeParseException {
+  QRCodeInvalidTotpSecretException() : super("invalid totp secret");
+}
+
+class QRCodeInvalidExpiryException extends QrCodeParseException {
+  QRCodeInvalidExpiryException() : super("invalid expiry date");
+}
+
+class QRCodeMissingExpiryException extends QrCodeFieldMissingException {
+  QRCodeMissingExpiryException() : super("expirationDate");
+}
+
+class QRCodeInvalidFormatException extends QrCodeParseException {
+  QRCodeInvalidFormatException() : super("invalid format");
 }
 
 class IdentificationQrContentParser {
   final CardDetailsModel _cardDetailsModel;
 
   IdentificationQrContentParser(this._cardDetailsModel);
 
-  QRCodeParseResult processQrCodeContent(String rawBase64Content) {
+  void processQrCodeContent(String rawBase64Content) {
+    final base64Decoder = Base64Decoder();
+
+    CardActivateModel cardActivateModel;
     try {
-      final base64Decoder = Base64Decoder();
+      var rawProtobufData = base64Decoder.convert(rawBase64Content);
+      cardActivateModel = CardActivateModel.fromBuffer(rawProtobufData);
+    } on Exception catch (_) {
+      throw QRCodeInvalidFormatException();
+    }
 
-      Uint8List rawProtobufData;
-      try {
-        rawProtobufData = base64Decoder.convert(rawBase64Content);
-      } on Exception catch (_) {
-        return QRCodeParseResult(
-            hasError: true,
-            userErrorMessage:
-                "Der Inhalt des eingescannten Codes kann nicht verstanden "
-                "werden. Vermutlich handelt es sich um einen QR Code, "
-                "der nicht für die Ehrenamtskarte App generiert wurde.");
-      }
-      CardActivateModel cardActivateModel;
-      try {
-        cardActivateModel = CardActivateModel.fromBuffer(rawProtobufData);
-      } on Exception catch (_) {
-        return QRCodeParseResult(
-            hasError: true,
-            userErrorMessage:
-                "Der Inhalt des eingescannten Codes kann nicht verstanden "
-                "werden. Vermutlich handelt es sich um einen QR Code, "
-                "der nicht für die Ehrenamtskarte App generiert wurde.");
-      }
+    final fullName = cardActivateModel.fullName;
+    if (fullName == null) {
+      throw QrCodeFieldMissingException("fullName");
+    }
+    final hashSecret = cardActivateModel.hashSecret;
+    if (hashSecret == null) {
+      throw QrCodeFieldMissingException("randomBytes");
+    }
```

Comment:
Do you know if there is a difference between `cardActivateModel.hasFullName()` and the null check we are currently doing? 
Should we also make simple length checks like size > 0?

Reply:
Also good points. I kept that logic just as it was, but makes sense to think about those thinks when already touching it.
- [ ] check difference `hasFullName()` and `== null`
- [ ] check for empty strings

Reply:
> Do you know if there is a difference between cardActivateModel.hasFullName() and the null check we are currently doing?

I looked it up. The `has` method for basic types also only checks for `null`. On lists, it also checks that the list is not empty.

I will use the `has` methods. Improves readability and they must be there for a reason 😄